integration/client.go

```diff
@@ -3,14 +3,41 @@
 package integration
 
 import (
+	giantclientset "github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	corev1alpha1 "github.com/giantswarm/apiextensions/pkg/clientset/versioned/typed/core/v1alpha1"
 	"github.com/giantswarm/microerror"
 	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	extensionsv1beta1 "k8s.io/client-go/kubernetes/typed/extensions/v1beta1"
 	"k8s.io/client-go/tools/clientcmd"
 
 	"github.com/giantswarm/e2e-harness/pkg/harness"
 )
 
-func getK8sClient() (kubernetes.Interface, error) {
+type k8sClient interface {
+	CoreV1() corev1.CoreV1Interface
+	ExtensionsV1beta1() extensionsv1beta1.ExtensionsV1beta1Interface
+	CoreV1alpha1() corev1alpha1.CoreV1alpha1Interface
```

Comment:
What will happen when we release `CoreV1`?

Reply:
I would iterate, applying the same approach again. At some point our `CoreV1` would differ from upstream's, there I would define the interface and an implementer type that decides which implementation to use for each of the requested methods.

Reply:
This way we can use a single client for interacting with k8s, which is good for the legibility and maintainability of code (the client code depend on who implements a resource IMO, maybe that can change in the future).